include/seqan/graph_types/graph_impl_directed.h

```diff
@@ -108,20 +159,34 @@ class Graph<Directed<TCargo, TSpec> >
 
 //////////////////////////////////////////////////////////////////////////////
 
-template<typename TCargo, typename TSpec>
-inline String<typename EdgeType<Graph<Directed<TCargo, TSpec> > >::Type*>&
-_getVertexString(Graph<Directed<TCargo, TSpec> > const& g) {
+template <typename TCargo, typename TSpec>
+inline String<typename EdgeType<Graph<Directed<TCargo, TSpec> > >::Type*> &
+_getVertexString(Graph<Directed<TCargo, TSpec> > const & g)
+{
     SEQAN_CHECKPOINT
     typedef Graph<Directed<TCargo, TSpec> > TGraph;
     typedef typename EdgeType<TGraph>::Type TEdgeStump;
     return const_cast<String<TEdgeStump*>&>(g.data_vertex);
 }
 
+//////////////////////////////////////////////////////////////////////////////
+
+template <typename TCargo>
+inline String<typename EdgeType<Graph<Directed<TCargo, WithSourceId> > >::Type*> &
+_getVertexInString(Graph<Directed<TCargo, WithSourceId> > const & g)
+{
+    SEQAN_CHECKPOINT
+    typedef Graph<Directed<TCargo, WithSourceId> > TGraph;
+    typedef typename EdgeType<TGraph>::Type TEdgeStump;
+    return const_cast<String<TEdgeStump*>&>(g.data_vertex_in);
+}
+
 /////////////////////////////////////////////////////////////////////////////
 
-template<typename TCargo, typename TSpec>
-inline typename VertexIdHandler<Graph<Directed<TCargo, TSpec> > >::Type&
-_getVertexIdManager(Graph<Directed<TCargo, TSpec> > const& g) {
+template <typename TCargo, typename TSpec>
+inline typename VertexIdHandler<Graph<Directed<TCargo, TSpec> > >::Type &
+_getVertexIdManager(Graph<Directed<TCargo, TSpec> > const & g)
+{
```

Comment:
Since you touched the evil interfaces :), could you please adapt them as described above.
Many thanks.